### PR TITLE
Bugfix - reveal validation

### PR DESCRIPTION
**Pull request by joefitter**
* Update form router to validate relevant option reveals
* Pass errors to connected child component

### CODE REVIEW

**Comment by lennym** [comment]
Any chance of some unit tests for the stuff in the form router?

**Comment by joefitter** [comment]
YES, #drunkentraincoding doesn't come with any guarantees

File: pages/common/routers/form.js [hunk 1]
```diff
@@ -27,6 +28,22 @@ if (process.env.CSRF === 'false') {
...
+    if (!field.options) {
+      return obj;
+    }
+    const selectedOptions = field.options.filter(opt => castArray(values[key]).includes(opt.value));
+    return selectedOptions.reduce((o, opt) => {
+      return {
+        ...o,
+        ...(opt.reveal || {})
+      };
+    }, {});
```
Comment (lennym) [inline pages/common/routers/form.js]: There's no mention of `obj` here, which makes me suspect that this won't work if there's more than one field in the schema which has reveal options.

I _think_ it needs this:

```suggestion
    }, obj);
```

Comment (lennym) [reply pages/common/routers/form.js]: Narrator voice: It did need that.

**Pull request merged**